channeld/commit_tx.c

```diff
@@ -285,6 +285,7 @@ struct bitcoin_tx *commit_tx(const tal_t *ctx,
 	 *    `dust_limit_satoshis`, add a [`to_remote`
 	 *    output](#to_remote-output).
 	 */
+	u8 *redeem = NULL;
```

Comment:
We never set default values, during declaration, rather we let it uninitialized until we actually decide whether it should be a fixed value or a computed value. In this case it'd be down in the `if-else` statement where we either set it to `NULL` or the actual redeem script.

This helps us pin down accidental paths with `valgrind`.

Reply:
fixed